full-service/src/db/txo.rs

```diff
@@ -1987,9 +2069,81 @@ impl TxoModel for Txo {
     }
 }
 
+fn add_authenticated_memo_to_database(
+    txo_id: &str,
+    sender_address_hash: &str,
+    payment_request_id: Option<i64>,
+    payment_intent_id: Option<i64>,
+    conn: Conn,
+) -> Result<(), WalletDbError> {
+    use crate::db::schema::authenticated_sender_memos;
+
+    let new_memo = NewAuthenticatedSenderMemo {
+        txo_id,
+        sender_address_hash,
+        payment_request_id,
+        payment_intent_id,
+    };
+
+    diesel::insert_into(authenticated_sender_memos::table)
+        .values(&new_memo)
+        .on_conflict_do_nothing()
+        .execute(conn)?;
+
+    Ok(())
+}
+
+fn i32_to_two_bytes(value: i32) -> [u8; 2] {
+    [(value >> 8) as u8, (value & 0xFF) as u8]
+}
+
+fn two_bytes_to_i32(bytes: [u8; 2]) -> i32 {
+    ((bytes[0] as i32) << 8) | (bytes[1] as i32)
+}
+
+fn add_memo_to_database(
+    txo_id: &str,
+    memo_payload: &MemoPayload,
+    conn: Conn,
+) -> Result<(), WalletDbError> {
+    // Interpret the memo payload and save it to the correct database table.
+    // Check that there is no existing memo before creating a new one.
+    match MemoType::try_from(memo_payload) {
+        Ok(MemoType::AuthenticatedSender(memo)) => add_authenticated_memo_to_database(
+            txo_id,
+            &memo.sender_address_hash().to_string(),
+            None,
+            None,
+            conn,
+        ),
+        Ok(MemoType::AuthenticatedSenderWithPaymentIntentId(memo)) => {
+            add_authenticated_memo_to_database(
+                txo_id,
+                &memo.sender_address_hash().to_string(),
+                None,
+                Some(memo.payment_intent_id() as i64),
+                conn,
+            )
+        }
+        Ok(MemoType::AuthenticatedSenderWithPaymentRequestId(memo)) => {
+            add_authenticated_memo_to_database(
+                txo_id,
+                &memo.sender_address_hash().to_string(),
+                Some(memo.payment_request_id() as i64),
+                None,
+                conn,
+            )
+        }
+        Ok(_) => Ok(()),
```

Comment:
When it's any of the other memo types available as defined in the mc transaction crate. We will eventually care about this information in a future update, but for now it is to be ignored. There will soon be a follow up PR that adds all of the DestinationMemo stuff, which would cause this to be updated